runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -98,43 +98,33 @@
      */
     public static class Index<K, V, I extends Comparable<?>> {
         private final CorfuTable.IndexName name;
-        private final CorfuTable.IndexFunction<K, V, I> indexFunction;
-        private final CorfuTable.MultiValueIndexFunction<K, V, I> multiValueIndexFunction;
+        private final CorfuTable.MultiValueIndexFunction<K, V, I> indexFunction;
 
-        // A flag representing whether a single indexer or a multi indexer
-        private final boolean monoIndex;
 
         public Index(CorfuTable.IndexName name, CorfuTable.IndexFunction<K, V, I> indexFunction) {
             this.name = name;
-            this.indexFunction = indexFunction;
-            this.multiValueIndexFunction =
+            this.indexFunction =
                     (k, v) -> Collections.singletonList(indexFunction.apply(k, v));
-            monoIndex = true;
         }
 
         public Index(CorfuTable.IndexName name,
                      CorfuTable.MultiValueIndexFunction<K, V, I> indexFunction) {
             this.name = name;
-            this.indexFunction = (k, v) -> indexFunction.apply(k, v).iterator().next();
-            this.multiValueIndexFunction = indexFunction;
-            monoIndex = false;
+            this.indexFunction = indexFunction;
         }
 
         public CorfuTable.IndexName getName() {
             return name;
         }
 
         public CorfuTable.IndexFunction<K, V, I> getIndexFunction() {
-            return indexFunction;
+            return (k, v) -> indexFunction.apply(k, v).iterator().next();
```

Comment:
Is it possible that `iterator` would be empty? If yes then you could get NPE

Reply:
Good point.